test/Interop/Cxx/symbolic-imports/indexing-emit-symbolic-module-interface.swift

```diff
@@ -116,31 +116,20 @@ import CxxModule
 // CHECK-EMPTY:
 // CHECK-NEXT:    init()
 // CHECK-EMPTY:
-// CHECK-NEXT:    init(y: Int32)
```

Comment:
Hmm, are we intentionally not emitting these constructors in the symbolic interface anymore?

Reply:
I think that would be the best approach 👍

Reply:
What is the symbolic interface used for?

Most declarations produced by ClangImporter were already marked as implicit, which seems correct for the purpose of the implicit bit. If I make the change to start printing synthesized declarations, are we going to over-print anything? Can we separate these changes, or do we think they must be made together?

Reply:
In C++ interop, the symbolic interface is used by SourceKit/Xcode for C++ class templates, since there isn't currently a mechanism that would instantiate C++ templates on the fly for the purpose of code completion, jump-to-definition, quick doc, etc.

Reply:
This logic is also used more generally for the generated module interface provided by SourceKit if you e.g jump-to-definition into an imported module. I'm not against doing the changes separately

Reply:
> are we going to over-print anything?

Jump-to-definition relies on the USR of a decl reference being present in the generated interface it jumps to, so I think we ought to have all synthesized decls printed, otherwise jump-to-definition would just open the module interface, which isn't too helpful. I believe this should only affect Clang cases since I don't think we ever synthesize public decls for Swift

Reply:
I explored distinguishing synthesized declarations from implicit but it seems like that might not really be the distinction we want as I believe that some forms of printing want to omit synthesized, too. I think the approach I'm going to take is to just include implicit decls in the symbolic interfaces for now.

Reply:
Actually, I've just landed on checking whether the implicit declaration belongs to a ClangModuleUnit.

Reply:
That doesn't work either. Now there are a bunch of other interface printing tests that are broken due to the inclusion of many unexpected synthesized clang declarations. I don't want to solve the subproblem of determining whether those are legitimate or not (some of them look a bit weird to me).

I think I'm going to scrap the whole approach of correctly marking the constructors as implicit and instead teach SILGen to avoid creating a profiler for decls belonging to a clang module, implicit or not. Someone more familiar with these various interface kinds should probably look into how best to make this a bit more self-consistent, though.